build/lib/src/state/filesystem_cache.dart

```diff
@@ -244,15 +245,18 @@ class InMemoryFilesystemCache implements FilesystemCache {
     Encoding encoding = utf8,
     required void Function() writer,
   }) {
-    _stringContentCache[id] = contents;
-    _bytesContentCache.remove(id);
-    _existsCache[id] = true;
-
+    // Encodings other than utf8 do not use `_stringContentCache`.
+    if (encoding == utf8) {
+      _stringContentCache[id] = contents;
+    } else {
+      _stringContentCache.remove(id);
+    }
```

Comment:
and this stuff could (almost) be replaced by a call to `writeAsBytes`.

Reply:
Thanks; extracted common `_writeBytes`.